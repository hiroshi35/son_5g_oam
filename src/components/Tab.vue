<template>
  <div  role="tabpanel" :class="{ 'tab-pane active': selected, 'tab-pane': !selected }" :id="id">
    <slot></slot>
  </div>
</template>

<script>
  export default {
    mounted: function () {
      this.pushTab()
    },
    props: ['title', 'selected'],
    computed: {
      id () {
        return this.title.replace(/[^a-zA-Z0-9]+/g, '_').toLowerCase()
      }
    },
    methods: {
      pushTab () {
        this.$nextTick(function () {
          this.$parent.tabs.push({
            id: this.id,
            title: this.title,
            selected: this.selected
          })
        })
      }
    }
  }
</script>

