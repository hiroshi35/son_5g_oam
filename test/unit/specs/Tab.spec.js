import { shallow } from 'vue-test-utils'
import Tab from '@/components/Tab.vue'

describe('Tab.vue', () => {
  let wrapper
  // Mock this.$parent.push()
  const mock = jest.fn()
  Tab.beforeCreate = function () { this.$parent = {tabs: {push: mock}} }
  beforeEach(() => {
    wrapper = shallow(Tab,
      {
        propsData: {
          title: 'Tab'
        },
        slots: {
          default: 'Test'
        }
      })
  })
  test('current tab should be pushed to parent tabs', () => {
    expect(mock.mock.calls.length).toBe(1)
    expect(mock.mock.calls[0][0].id).toBe('tab')
    expect(mock.mock.calls[0][0].title).toBe('Tab')
  })
  it('should be vue instance', () => {
    expect(wrapper.isVueInstance()).toBeTruthy()
  })
  it('has the expected html structure', () => {
    expect(wrapper.element).toMatchSnapshot()
  })
  it('has correct props', () => {
    expect(wrapper.props().title).toBe('Tab')
  })
})
