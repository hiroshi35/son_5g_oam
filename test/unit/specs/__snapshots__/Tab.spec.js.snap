// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tab.vue has the expected html structure 1`] = `
<div
  class="tab-pane"
  id="tab"
  role="tabpanel"
>
  Test
</div>
`;
